src/python/pants/backend/jvm/subsystems/jvm_platform.py

```diff
@@ -257,17 +271,33 @@ class JvmPlatformSettings:
     class IllegalSourceTargetCombination(TaskError):
         """Illegal pair of -source and -target flags to compile java."""
 
-    def __init__(self, source_level, target_level, args, name=None, by_default=False):
+    @staticmethod
+    def _copy_as_default(original, name):
+        """Copies the original with a new name, setting by_default to True."""
+        return JvmPlatformSettings(
+            source_level=original.source_level,
+            target_level=original.target_level,
+            args=original.args,
+            jvm_options=original.jvm_options,
+            name=name,
+            by_default=True,
+        )
+
+    def __init__(
+        self, *, source_level, target_level, args, jvm_options, name=None, by_default=False
+    ):
         """
     :param source_level: Revision object or string for the java source level.
     :param target_level: Revision object or string for the java target level.
     :param list args: Additional arguments to pass to the java compiler.
+    :param list jvm_options: Additional jvm options specific to this JVM platform.
     :param str name: name to identify this platform.
     :param by_default: True if this value was inferred by omission of a specific platform setting.
     """
         self.source_level = JvmPlatform.parse_java_version(source_level)
         self.target_level = JvmPlatform.parse_java_version(target_level)
-        self.args = tuple(args or ())
+        self.args = tuple(flatten_shlexed_list(args or ()))
```

Comment:
Wondering if using `FlattenedOrderedSet` would make sense here. Context: https://github.com/pantsbuild/pants/pull/9229

Reply:
I don't think so. The insert order of the commandline args matters and I don't think you want to dedup them either.